Java 11

darkBasic666 opened the pull request:
## Summary
- Update java version to 11
- Add new GC's Options to README

Commit:
update java version

Commit:
fix tests

Commit:
Fix Tests and add new GCs options to README

zeminlu commented on build.gradle:
```diff
@@ -1,6 +1,6 @@
 buildscript {
     ext {
-        springBootVersion = '2.0.6.RELEASE'
+        springBootVersion = '2.1.0.RELEASE'
```
Any reason why we shouldn't be upgrading to a more recent version? I believe 2.2.6 is the latest. @leandrocostawolox any thoughts on this?

leandrocostawolox replied:
Totally! Let's use 2.2.6
2.1.0 it's from 2018

zeminlu commented on src/test/java/wolox/bootstrap/BootstrapApplicationTests.java:
```diff
@@ -3,10 +3,14 @@
 
 @RunWith(SpringRunner.class)
 @SpringBootTest
+@PropertySource(value = "classpath:application.properties")
```
Is this necessary? Doesn't springboot already fetch this one automatically?

Jimenarosello replied:
I just checked this, the application context doesn't load without this annotation :(

leandrocostawolox replied:
Probably it's necessary because we are not using the Spring Boot v2.2.6
Let's try to delete this line after @darkBasic666 update the Spring Boot version :)

zeminlu commented on README.md:
````diff
@@ -120,6 +120,25 @@ spring.datasource.username=DBUSERNAME
 spring.datasource.password=DBPASSWORD
 ```
 
+#### Garbage Collection
````
EXCELENT!!

zeminlu commented on README.md:
````diff
@@ -120,6 +120,25 @@ spring.datasource.username=DBUSERNAME
 ```
 
+#### Garbage Collection
+In Java 11 added news Garbage Collection options to measuring, managing and better performance. 
````
```suggestion
In Java 11, new Garbage Collection options have been added for measuring and managing this process, as well as for enhanced performance. 
```

zeminlu commented on README.md:
```diff
@@ -120,6 +120,25 @@ spring.datasource.username=DBUSERNAME
 
+#### Garbage Collection
+In Java 11 added news Garbage Collection options to measuring, managing and better performance. 
+1. Epsilon : Epsilon removes the impact GC has on performance. You can use it to see how garbage collection affects your app's performance and what your memory threshold is since it'll tell you when it runs out.
```
```suggestion
1. Epsilon : Epsilon removes the impact that the GC has on performance. You can use it to see how garbage collection affects your app's performance and what your memory threshold is, since it'll tell you when it runs out.
```

zeminlu commented on README.md:
```diff
@@ -120,6 +120,25 @@ spring.datasource.username=DBUSERNAME
+In Java 11 added news Garbage Collection options to measuring, managing and better performance. 
+1. Epsilon : Epsilon removes the impact GC has on performance. You can use it to see how garbage collection affects your app's performance and what your memory threshold is since it'll tell you when it runs out.
+
+2. ZGC (Z Garbage Collector) : ZGC's design works well with applications large heap sizes. It manages these heaps with pause times under 10ms and little impact on throughput. These times are better than G1's.
```
```suggestion
2. ZGC (Z Garbage Collector) : ZGC's design works well with applications with large heap sizes. It manages these heaps with pause times under 10ms and with little impact on throughput. These times are better than those of G1's.
```

zeminlu commented on src/main/resources/application.properties:
```diff
@@ -9,3 +9,4 @@ spring.jpa.hibernate.ddl-auto=update
 spring.jpa.properties.hibernate.jdbc.lob.non_contextual_creation=true
 spring.messages.basename=messages
 log.file.output=application.log
+spring.main.allow-bean-definition-overriding=true
```
Please add in the PR description a small explanation why this option is needed.

leandrocostawolox replied:
Remove. Not needed

zeminlu commented on build.gradle:
```diff
@@ -37,4 +38,12 @@ dependencies {
     annotationProcessor('org.projectlombok:lombok:1.18.12')
     compile('org.springframework.boot:spring-boot-starter-hateoas')
     compile('org.simplejavamail:simple-java-mail:4.1.3')
+    compile('javax.xml.bind:jaxb-api:2.3.0')
```
Please add in the PRs description the reasons for adding these dependencies.

leandrocostawolox replied:
Remove. Not needed

zeminlu reviewed (changes requested):
Awesome work! Check my comments please :)

Jimenarosello committed:
Apply suggestions from code review

Phrasing corrections

Co-authored-by: Luciano Zemin <zeminlu@wolox.com.ar>

JimeRosello committed:
Remove unnecessary property source annotation

JimeRosello committed:
Revert last commit

leandrocostawolox commented on src/test/resources/application.properties:
```diff
@@ -7,3 +7,4 @@ spring.datasource.username=sa
 spring.datasource.password=sa
 log.file.output=application.log
 spring.jpa.show-sql=true
+spring.main.allow-bean-definition-overriding=true
```
Add end of line at the end of this line

leandrocostawolox replied:
Done!

leandrocostawolox reviewed (changes requested):
Nice work! I wrote you some comments :) 
I was thinking that maybe we can add some features in test classes (So we don't have to change the application logic itself):
- Use var (Type inference) [link](https://www.baeldung.com/java-10-local-variable-type-inference)
- Show some new Optional methods (`ifPresentOrElse`, `or`)

Commit:
Migrate to Spring Boot 2.2.6

- Fix URIs to avoid extra /
- Remove PropertySource
- Exclude JUnit Vintage from build.gradle
- Remove unused dependencies
- Fix repository methods which didn't work

Commit:
Fix test according with Spring Boot 2.2.6

- Use only JUnit 5
- Add spring security test annotations
- Apply linter suggestions
- Fix some broken tests

leandrocostawolox committed:
Add API constants class and use controller URI constants

leandrocostawolox committed:
Use ApiConstants in tests + Use tests naming convention

leandrocostawolox committed:
Extract base API URI to constant + Rename to APIConstants

leandrocostawolox committed:
Merge pull request #39 from Wolox/java-11/spring-boot-2.2.6

Java 11 - Migration to Spring Boot 2.2.6

leandrocostawolox committed:
Using java version 11 constant in build.gradle

leandrocostawolox committed:
Adding var uses in RoleController and CustomUserDetailService

leandrocostawolox committed:
Two linter fixes

leandrocostawolox commented:
Merge already reviewed branch spring-boot-2.2.6 + Add 5 basic commits (after this merge) to review :)

leandrocostawolox committed:
Add junit platform to build.gradle to run tests using JUnit 5

leandrocostawolox committed:
Add orElseThrow method instead of get
